src/context/ast/QueryAstContext.h

```diff
@@ -1,188 +1,59 @@
-    PaginationContext() : CypherClauseContextBase(CypherClauseKind::kPagination) {}
-
-    int64_t     skip{0};
-    int64_t     limit{std::numeric_limits<int64_t>::max()};
-};
-
-struct YieldClauseContext final : CypherClauseContextBase {
-    YieldClauseContext() : CypherClauseContextBase(CypherClauseKind::kYield) {}
-
-    bool                                              distinct{false};
-    const YieldColumns*                               yieldColumns{nullptr};
-    std::unordered_map<std::string, AliasType>*       aliasesUsed{nullptr};
-
-    bool                                              hasAgg_{false};
-    bool                                              needGenProject_{false};
-    YieldColumns*                                     projCols_;
-    std::vector<Expression*>                          groupKeys_;
-    std::vector<Expression*>                          groupItems_;
-    std::vector<std::string>                          aggOutputColumnNames_;
-    std::vector<std::string>                          projOutputColumnNames_;
-};
-
-struct ReturnClauseContext final : CypherClauseContextBase {
-    ReturnClauseContext() : CypherClauseContextBase(CypherClauseKind::kReturn) {}
-
-    std::unique_ptr<OrderByClauseContext>        order;
-    std::unique_ptr<PaginationContext>           pagination;
-    std::unique_ptr<YieldClauseContext>          yield;
-};
-
-struct WithClauseContext final : CypherClauseContextBase {
-    WithClauseContext() : CypherClauseContextBase(CypherClauseKind::kWith) {}
-
-    std::unique_ptr<OrderByClauseContext>       order;
-    std::unique_ptr<PaginationContext>          pagination;
-    std::unique_ptr<WhereClauseContext>         where;
-    std::unique_ptr<YieldClauseContext>         yield;
-    std::unordered_map<std::string, AliasType>  aliasesGenerated;
-};
-
-struct MatchClauseContext final : CypherClauseContextBase {
-    MatchClauseContext() : CypherClauseContextBase(CypherClauseKind::kMatch) {}
-
-    std::vector<NodeInfo>                       nodeInfos;
-    std::vector<EdgeInfo>                       edgeInfos;
-    std::unique_ptr<PathBuildExpression>        pathBuild;
-    std::unique_ptr<WhereClauseContext>         where;
-    std::unordered_map<std::string, AliasType>* aliasesUsed{nullptr};
-    std::unordered_map<std::string, AliasType>  aliasesGenerated;
-};
-
-struct UnwindClauseContext final : CypherClauseContextBase {
-    UnwindClauseContext() : CypherClauseContextBase(CypherClauseKind::kUnwind) {}
-
-    const YieldColumns*                         yieldColumns{nullptr};
-    std::unordered_map<std::string, AliasType>* aliasesUsed{nullptr};
-    std::unordered_map<std::string, AliasType>  aliasesGenerated;
+enum FromType {
+    kInstant,
+    kVariable,
+    kPipe;
 };
 
-struct MatchAstContext final : AstContext {
-    // Alternative of Match/Unwind/With and ends with Return.
-    std::vector<std::unique_ptr<CypherClauseContextBase>>  clauses;
+struct Starts {
```

Comment:
Where use these structure? I think you'd better add them when use it.

Reply:
Avoid too much content in one PR， will use it in next PR